qcodes/dataset/measurements.py

```diff
@@ -639,10 +641,21 @@ def __exit__(self, exception_type, exception_value, traceback) -> None:
         for func, args in self.exitactions:
             func(*args)
 
+        if exception_type:
+            # if an exception happened during the measurement,
+            # log the exception
+            stream = io.StringIO()
+            tb_module.print_exception(exception_type,
+                                      exception_value,
+                                      traceback,
+                                      file=stream)
+            log.warning('An exception occured in measurement with guid: '
+                        f'{self.ds.guid}; Traceback: {stream.getvalue()}')
```

Comment:
newlines for readability? :)
```suggestion
                        f'{self.ds.guid};\nTraceback:\n{stream.getvalue()}')
```